src/Microsoft.VisualStudio.LanguageServices.Razor/ProjectSystem/DefaultRazorProjectHost.cs

```diff
@@ -2,6 +2,8 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
```

Comment:
- tl;dr; I'd recommend reviewing this class as an entire file instead of looking at the diff.

Had to refactor this class in VS4Mac so to keep both worlds consistent I refactored the windows one as well. In some places I'm extra verbose/protective to stay consistent across platforms. In the end it came out a lot more clear and in was able to unit test 99% of it. Ported the VS4Mac unit tests on this refactored guy as well and left the existing tests to ensure everything worked 100% the same.

Reply:
This looks good.